### PR TITLE
CMake: Split the main CMakeLists.txt into separate files

### PR DESCRIPTION
**Pull request by pcercuei**
Plus some cleanups.

Signed-off-by: Paul Cercueil <paul@crapouillou.net>

### CODE REVIEW

**Comment by afritz1** [comment]
In ` OpenTESArena/CMakeLists.txt`, shouldn't the word `TESArena` be `OpenTESArena`? I'm not sure why it was shortened in the first place.

**Comment by Ragora** [comment]
I shortened it to prevent a naming clash with the OpenTESArena folder name.

**Comment by afritz1** [comment]
Okay. In release packages, though, I'd like the executable to use the full name.

**Comment by Ragora** [comment]
You'd have to change the folder name or change casing somewhere (which I personally wouldn't particularly like).

**Comment by afritz1** [comment]
I mean for the `.tar.gz` release package that would just have the executable and other assorted files in it, so people are running `OpenTESArena` instead of `TESArena`. This wouldn't be a part of the build itself; just what people get at the end.

**Comment by Ragora** [comment]
The thing I linked would be a tool to generate those automatically (which also includes App Bundles, RPM's and Debs) from the CMake projects. That's why I brought it up.

**Comment by afritz1** [comment]
Oh, okay. We can use package creation tools at some point then.

**Comment by Ragora** [comment]
I haven't used CPack myself yet, I'm sure someone would have a better idea of its any good.

File: components/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+PROJECT(components CXX)
+
+FILE(GLOB COMP_SOURCES vfs/*.h vfs/*.hpp vfs/*.cpp
+	archives/*.h archives/*.hpp archives/*.cpp
+)
+
+IF (WIN32)
+	FILE(GLOB COMP_MISC_SOURCES misc/*.h misc/*.hpp misc/*.cpp)
```
Comment (pcercuei) [inline components/CMakeLists.txt]: OOps, there's a bug here. Sorry about that.
It should be COMP_WIN32_SOURCES, which is used line 11.

**Pull request merged**